### ISSUE TITLE
Add possibility to define custom highlighter

### ISSUE THREAD

**Issue by vladshcherbin** [scriptcoded/sql-highlight#6]
👋 

Hey, in a similar library I've used a feature where you could define your custom highlighter with regex and color, something like this:

```js

own: {
  duration: {
    regexp: /^(?:(?! \|).)*/,
    color: 'magenta'
  }
}
```

It gives possibility to color custom parts of query, for example if it includes query duration.

Would be nice to have something similar here later if possible :)

**Comment by scriptcoded** [comment]
This would absolutely be a useful feature to have. One major issue with the library at the moment is that it's using plain regex for highlighting. While this works fine in most cases, more complex queries and language features might not be properly highlighted, and even break the highlighter as a whole. I think a general improvement for this library would be to look in to alternative methods for matching language features, and perhaps then this would make more sense. I think this feature was to be implemented right now we would also have to provide some way of choosing what position in [`highlighters`](https://github.com/scriptcoded/sql-highlight/blob/80bb46617704e87c2d4cfa54ccb7ecd8dab1a486/index.js#L21) the custom regex should end up.

As for this specific case, perhaps the rule is something that should be put into the base library?

**Comment by vladshcherbin** [comment]
@scriptcoded yeah, I understand the difficulty with where you execute custom regex. Maybe it can be set with some kind of an option (order):

```
duration: {
  regexp: /^(?:(?! \|).)*/,
  color: 'magenta',
  order: 'last'
}
```

`last`, `first`, `{ after: 'keyword' }` - maybe this can be values 🤔 

Previously, I was using [igniculus](https://github.com/Undre4m/igniculus) and this idea (and other my issues) come from there. [Custom rules section](https://github.com/Undre4m/igniculus#custom-rules) there also has a warning mark as an experimental advanced feature.

As per my case, I was using this regex to highlight query duration `10.12 ms |`, don't think this should be added to base library as format differs according to every dev preferences 🤔

**Comment by scriptcoded** [comment]
@vladshcherbin I like the idea of using `order`. The issue with collisions remains, but I think adding a warning like igniculus did would be enough to warn people. I'll mock something up.

**Comment by scriptcoded** [comment]
Closed in #57! 🎉
